src/createStore.ts

```diff
@@ -39,35 +39,54 @@ import { kindOf } from './utils/kindOf'
  * @returns A Redux store that lets you read the state, dispatch actions
  * and subscribe to changes.
  */
+
+// 3 overloaded functions
+// 1st overload
+//     v--'defualt' allows createStore function to be called by different name when imported later 
 export default function createStore<
-  S,
-  A extends Action,
-  Ext = {},
-  StateExt = never
+  S, // generic of type PreloadedState found in the sotre.ts file and defined below (let currentState = preloadedState as S)
```

Comment:
Just a commentary on generics and readability. I think that we do ourselves a diservice when we give generics single letter / cryptic aliases. Why not call this generic `PreloadedState`? I feel like readability is important in any codebase, but especially in opensource!

Reply:
I completely agree! I don't like to have to look back at other parts of the code to know what a variable represents.